test/watch.js

```diff
@@ -0,0 +1,441 @@
+console.log("hello world")
+});`);
+      fileContains(t, 'public/index.html', '<h1>hello world</h1>');
+      fileContains(t, 'public/app.js', `require.register("new-file.js", function(exports, require, module) {
+console.log("new")
+});`);
+      t.end();
+    } else {
+      t.fail(`Unexpected number of calls ${calls}`);
+    }
+  };
+
+  watch({}, onCompile);
+});
+
+test.serial.cb('detect file removal', t => {
+  fixturify.writeSync('.', {
+    'brunch-config.js': `module.exports = {
+      files: {
+        javascripts: {
+          joinTo: 'app.js'
+        }
+      }
+    };`,
+    'app': {
+      'assets': {
+        'index.html': '<h1>hello world</h1>'
+      },
+      'a.js': 'filea',
+      'b.js': 'fileb'
+    }
+  });
+
+  var calls = 0;
+
+  const onCompile = function() {
+    calls++;
+
+    if (calls === 1) {
+      fileExists(t, 'public/app.js.map');
+      fileContains(t, 'public/app.js', '//# sourceMappingURL=app.js.map');
+      fileContains(t, 'public/app.js', `require.register("a.js", function(exports, require, module) {
+filea
+});`);
+      fileContains(t, 'public/app.js', `require.register("b.js", function(exports, require, module) {
+fileb
+});`);
+      fileContains(t, 'public/index.html', '<h1>hello world</h1>');
+
+      fs.unlinkSync('app/b.js');
+    } else if (calls === 2) {
+      fileExists(t, 'public/app.js.map');
+      fileContains(t, 'public/app.js', '//# sourceMappingURL=app.js.map');
+      fileContains(t, 'public/app.js', `require.register("a.js", function(exports, require, module) {
+filea
+});`);
+      fileDoesNotContains(t, 'public/app.js', `require.register("b.js", function(exports, require, module) {
+fileb
+});`);
+      fileContains(t, 'public/index.html', '<h1>hello world</h1>');
+      t.end();
+    } else {
+      t.fail(`Unexpected number of calls ${calls}`);
+    }
+  };
+
+  watch({}, onCompile);
+});
+
+test.serial.cb('install npm packages if package.json changes', t => {
+  fixturify.writeSync('.', {
+    'brunch-config.js': `module.exports = {
+      files: {
+        javascripts: {
+          joinTo: 'app.js'
+        }
+      }
+    };`,
+    'app': {
+      'assets': {
+        'index.html': '<h1>hello world</h1>'
+      },
+      'initialize.js': 'console.log("hello world")'
+    }
+  });
+
+  var calls = 0;
+
+  const onCompile = function() {
+    calls++;
+
+    if (calls === 1) {
+      t.true(fs.readdirSync('./node_modules').indexOf('lodash') === -1);
+
+      const packageJson = JSON.parse(fs.readFileSync('package.json', 'utf8'));
+      packageJson.dependencies.lodash = '*';
+      fs.writeFileSync('package.json', JSON.stringify(packageJson, null, 2));
+    } else if (calls === 2) {
+      t.true(fs.readdirSync('./node_modules').indexOf('lodash') !== -1);
+      t.end();
+    } else {
+      t.fail(`Unexpected number of calls ${calls}`);
+    }
+  };
+
+  watch({}, onCompile);
+});
+
+test.serial.cb('install bower components if bower.json changes', t => {
```

Comment:
I'm testing any feature provided by brunch today. I'm also using an application with brunch and bower, because in the time I started it, brunch didn't have the npm integration and I didn't find a time to migrate. If we want to discourage your usage, I think we should add some warning. Also npm is known to be slower, because this I believe that some people are still using bower, even it having your own issues.